client/web/src/global/CoolCodeIntel.tsx

```diff
@@ -0,0 +1,842 @@
+    !isErrorLike(settingsCascade.final) && settingsCascade.final?.experimentalFeatures?.coolCodeIntel !== false
+
+export const CoolCodeIntel: React.FunctionComponent<CoolCodeIntelProps & { onClose: () => void }> = props => {
+    if (!props.coolCodeIntelEnabled) {
+        return null
+    }
+
+    return <CoolCodeIntelResizablePanel {...props} />
+}
+
+const LAST_TAB_STORAGE_KEY = 'CoolCodeIntel.lastTab'
+
+type CoolCodeIntelTabID = 'references' | 'token' | 'definition'
+
+interface CoolCodeIntelTab {
+    id: CoolCodeIntelTabID
+    label: string
+    component: React.ComponentType<CoolCodeIntelProps>
+}
+
+export const ReferencesPanel: React.FunctionComponent<CoolCodeIntelProps> = props => {
+    if (!props.clickedToken) {
+        return null
+    }
+
+    return <ReferencesList clickedToken={props.clickedToken} {...props} />
+}
+
+interface Location {
+    resource: {
+        path: string
+        content: string
+        repository: {
+            name: string
+        }
+        commit: {
+            oid: string
+        }
+    }
+    range?: {
+        start: {
+            line: number
+            character: number
+        }
+        end: {
+            line: number
+            character: number
+        }
+    }
+
+    url: string
+    lines: string[]
+}
+
+const buildLocation = (node: LocationFields): Location => {
+    const location: Location = {
+        resource: {
+            repository: { name: node.resource.repository.name },
+            content: node.resource.content,
+            path: node.resource.path,
+            commit: node.resource.commit,
+        },
+        url: '',
+        lines: [],
+    }
+    if (node.range !== null) {
+        location.range = node.range
+    }
+    location.url = buildFileURL(location)
+    location.lines = location.resource.content.split(/\r?\n/)
+    return location
+}
+
+interface RepoLocationGroup {
+    repoName: string
+    referenceGroups: LocationGroup[]
+}
+
+interface LocationGroup {
+    repoName: string
+    path: string
+    locations: Location[]
+}
+
+export const ReferencesList: React.FunctionComponent<
+    {
+        clickedToken: CoolClickedToken
+    } & Omit<BlobProps, 'className' | 'wrapCode' | 'blobInfo' | 'disableStatusBar'>
+> = props => {
+    const [activeLocation, setActiveLocation] = useState<Location | undefined>(undefined)
+    const [filter, setFilter] = useState<string | undefined>(undefined)
+    const debouncedFilter = useDebounce(filter, 150)
+
+    useEffect(() => {
+        setActiveLocation(undefined)
+        setFilter(undefined)
+    }, [props.clickedToken])
+
+    const history = useMemo(() => H.createMemoryHistory(), [])
+
+    const onReferenceClick = (location: Location | undefined): void => {
+        if (location) {
+            history.push(location.url)
+        }
+        setActiveLocation(location)
+    }
+
+    return (
+        <>
+            <input
```

Comment:
There is `Input` now :) (make sure to drop the className form-control, it's automatically added now.
```suggestion
            <Input
```